doc/Programming-guide/Key-platform-features/Common-Type-Library/index.md

```diff
@@ -33,38 +72,77 @@ Matching FQN in a different Tenant | N/A | OK | OK
-  "name" : "SampleCT",
-  "namespace" : "org.kaaproject.sample",
-  "version" : 1,
-  "dependencies" : [{ "fqn": "org.kaaproject.sample.ReferencedCT", "version" : 2}]
-  "fields" : [
-    ...
-  ]
-}
-```
+In **Add new type** window fill in all the required fields and click **Add** to apply the changes.
+Below is shown an example of **Add new type** window for Tenant CT.
+![Create Tenant CTL](attach/create_tenant_ctl.png)
+There is also ability to create a new version of CT by clicking **Create new version** at the top of **CT details** window which is described in the next chapter.
+
+## Get or view CT details
+
+CT details can be obtained via [REST API]({{root_url}}Programming-guide/Server-REST-APIs#TODO) call or can be viewed in **CT details** window of Administration UI.
+To open CT details click the row of the CTs list in corresponding CTL window.
+Below is shown an example of **CT details** window for Application CT.
+![View Application CTL](attach/view_application_ctl.png)
+To view another version of the CT, click on the version drop-down and select the appropriate version.
+
+## Delete CT schema
+
+CT schema can be deleted via [REST API]({{root_url}}Programming-guide/Server-REST-APIs#TODO) call or from **CT details** window of Administration UI.
+To delete CT schema open **CT details** and click **Delete** button located in the top of the window.
+
+> **NOTE:** The Kaa admin can delete System CT schema.
+The Tenant admin can delete Tenant CT schema.
+The Tenant developer can delete Application CT schema. 
+
+## CT promotion
+
+Users are able to promote the CT's scope from APPLICATION to TENANT, provided that there is no other CT with the identical FQN within the given tenant, and that they have a permission to do so.
+Otherwise an appropriate error will be returned.
+Promotion procedure updates scope of CT including all its versions.
+Promotion of CT can be performed in two ways: via [REST API]({{root_url}}Programming-guide/Server-REST-APIs#TODO) call or by clicking **Promote** in the **Application CTL** window of Administration UI as shown below.
+![Promote CTL](attach/application_ctl_promote.png)
 
-The CTL user interface automatically suggests the next available version (max loaded + 1) for a given FQN when creating a new schema.
-When referencing CTs, the user must specify both the FQN and the version. CTs may reference each other (see org.kaaproject.sample.ReferencedCT in the example above). Cyclic dependencies are not permitted. (Thus, CTs are nodes in a directed acyclic graph of dependencies.)
+## CT schema import and export
 
+The import of CT schema can be done in two ways: 
 
-## CT schema export support ##
+* perform [REST API]({{root_url}}Programming-guide/Server-REST-APIs#TODO) call. 
+* add new CT in Administration UI and upload schema json file as shown below.
+![CTL Import](attach/ctl_import.png)
 
 There are four ways of CT schema export:
 
 1.	"**shallow**" export simply retrieves the given schema as a file.
```

Comment:
According to the Documentation contribution guide, this list should be of bullet type rather than enumerated.